### ISSUE TITLE
Bidirectional recurrent units

### ISSUE THREAD

**Issue by amishgoel** [fchollet/keras#3281]
Hi. I am trying to implement a birdirectional recurrent network. As there is an option in keras for recurrent model, to set go_backwards =True, to process the input sequence backwards. As far I can tell from the source, it uses theano_scan go_backwards flag. From theano implementation, I think it  processes the input backwards, but does not flip the output along the timestep dimension. Generally, such reversal is important in most bidirectional implementation. I would like to confirm if the output is flipped in time or not while using go_backwards in keras. (One can use Lambda layer to flip later but it would be nice if it is explicit in documentation like some other alternate libraries such as lasagne.)

**Comment by farizrahman4u** [comment]
You can use bidirectional wrapper from seq2seq. Just put `Bidirectional()` around your RNN:

```python
from seq2seq.layers import Bidirectional
model = Sequential()
model.add(Bidirectional(LSTM(20, input_dim=30))
```

**Comment by amishgoel** [comment]
OK.. but still please confirm what does go_backwards produce... I am also
interested in implementing the logic of bidirectional networks myself from
learning point of view.

**Comment by farizrahman4u** [comment]
go_backwards simply scans in the opposite direction . It doen't flip the output, which means your mask and sequence will be out of alignment.

**Comment by amishgoel** [comment]
Thanks, that clarifies my doubt.
